app/policies/user_policy.rb

```diff
@@ -14,4 +14,8 @@ def search?
   def show?
     user&.treasurer? || record == user
   end
+
+  def activities?
+    user&.treasurer? || record == user
```

Comment:
Activities is a plural right? So it show multile records, in that case record==user should not work. I think you should create a scope which limits the records to the current user records

Reply:
We've worked on it together, yesterday. We could not figure out a perspective where using a policy Scope could work.

The reasoning is this:
- You want all activities where queried user (say: Alice) has orders
- The querying user (say: Bob) can only see his own orders, except when Bob is a Treasurer
- A scope would imply that Bob automatically gets his own (👍 ), but a treasurer would receive ALL orders instead of just Alices' (👎 ).
- So: a scope would return too much orders

@wilco375 maybe, just maybe, using the already available scope from Orders (`policy_scope(Order)`) and then further narrow the query down could work

Reply:
I'm thinking:
```ruby
Activity.select(%i[id title start_time]).joins(:orders).merge(policy_scope(Order).where(user: user).distinct
```

I don't know if the syntax is totally correct, though...

Reply:
@wilco375 Well I was actually waiting for a comment on this one ;)

Reply:
This actually is okay this way. The policy is about the user (since it is a user endpoint) so this is correct, I misunderstood in the first place